### DISCUSSION TITLE
Error: "Too many LUA scripts"

### DISCUSSION THREAD

**Discussion by jonathanthompson-me** [EdgeTX/edgetx#5383] [Q&A] [answered]
Today when playing with led color settings under global functions on the MT12 I started getting a "Too Many LUA Scripts" error message. I rebuilt the main branch on GitPod and updated the firmware to verify and it still happens. There's only two RGBLED lines in the global functions page. A third one results in that error. No other settings in that page.
Any ideas how I need to troubleshoot this error?

**Comment by jonathanthompson-me** [comment]
I have 7 entries in the Special Functions on the model, 5 of which are RGBLED entries. Looking in the code I guess the MAX_SCRIPTS is either 7 or 9. Maybe that's what the story is.

**Reply by pfeerick** [reply, marked as answer]
For MT12 it would be 7. There is a count of how many Lua SFs are defined (thus 2 in GF, plus 5 in SF = 7, so the 8th is over the limit), and and the radio will show that error/warning when you try to exceed the limit for that radio. Not sure if it takes into account if the line is enabled or not. Any particular why so many, rather than a single Lua that can do any conditional stuff itself?

**Reply by jonathanthompson-me** [reply]
No specific reason, just trying stuff to learn more about EdgeTX. I haven't got into writing lua scripts just yet, but that's coming shortly.
